src/Paprika/Store/FanOutList.cs

```diff
@@ -85,4 +85,20 @@ public void Accept(IPageVisitor visitor, IPageResolver resolver)
             }
         }
     }
+    public void Destroy(IBatchContext batch, in NibblePath prefix)
+    {
+        // Destroy the Id entry about it
+        Set(prefix, ReadOnlySpan<byte>.Empty, batch);
+
+        // Destroy the account entry
+        // SetAtRoot<FanOutPage>(batch, account, ReadOnlySpan<byte>.Empty, ref Data.StateRoot);
+
+        // Remove the cached
+        // batch.IdCache.Remove(prefix.UnsafeAsKeccak);
+
+        var index = GetIndex(prefix);
+        var addr = _addresses[index];
+
+        TPage.Wrap(batch.GetAt(addr)).Destroy(batch, prefix.SliceFrom(ConsumedNibbles));
```

Comment:
The page may COW. You should `batch.GetAddress` after destroy and update accordingly. If you `ref var addr =  ref  _addresses[index]` you should be able to assign it.

Reply:
Have tried addressing this. Can you confirm if the latest changes look right?